### PR TITLE
ADBDEV-4847: Clean up segments helper processes after error during restore

### PR DESCRIPTION
**Pull request by RekGRpth**
Clean up segments helper processes after error during restore.

The data.restoreDataFromTimestamp function launches helper processes on segments
during a single data file restore or during a resize restore. At the same time,
the restore.DoCleanup function cleans these processes only in a single data file
restore, and does not clean them in a resize restore. This can result in helper
processes on segments being abandoned after error during restore. This patch
solves this problem by causing cleanup in the second condition as well. For
testing, a test backup was prepared, which is restored with an error.

### CODE REVIEW

**Comment by InnerLife0** [comment]
> next line contains GetBackupFPInfoListFromRestorePlan function, that requires backupConfig to be not nil

~~The evil of global variables~~

Thanks

File: end_to_end/end_to_end_suite_test.go [hunk 1]
```diff
@@ -2254,6 +2254,20 @@ LANGUAGE plpgsql NO SQL;`)
...
+			command := exec.Command("tar", "-xzf", "resources/4-segment-db-error.tar.gz", "-C", backupDir)
+			mustRunCommand(command)
+			gprestoreCmd := exec.Command(gprestorePath,
+				"--timestamp", "20240226140135",
+				"--redirect-db", "restoredb",
+				"--backup-dir", path.Join(backupDir, "4-segment-db-error"),
+				"--resize-cluster")
+			output, err := gprestoreCmd.CombinedOutput()
+			Expect(err).To(HaveOccurred())
+			Expect(string(output)).To(ContainSubstring(`Error loading data into table public.test: COPY test, line 8193, column test: "32768": ERROR: value "32768" is out of range for type smallint`))
```
Comment (InnerLife0) [inline end_to_end/end_to_end_suite_test.go:2267]: If I understand this right, backup contains 4 data files for restored table. Each of them has values which are out of range. This values are different in each file.
As files are restored in parallel, the message can contain different values (not only 32768).
This test case in stable flaky on not patched version on my machine.

Comment (RekGRpth) [reply end_to_end/end_to_end_suite_test.go]: fixed

**Review by InnerLife0 — APPROVED** [review]

**Comment by RekGRpth** [comment]
bender build

**Pull request merged**